include/zephyr/drivers/stepper.h

```diff
@@ -71,20 +71,23 @@ enum stepper_run_mode {
 	/** Hold Mode */
 	STEPPER_HOLD_MODE = 0,
 	/** Position Mode*/
-	STEPPER_POSITION_MODE,
+	STEPPER_POSITION_MODE = 1,
 	/** Velocity Mode */
-	STEPPER_VELOCITY_MODE,
+	STEPPER_VELOCITY_MODE = 2,
 };
 
 /**
- * @brief Stepper Motor signal results
+ * @brief Stepper Motor Events
  */
-enum stepper_signal_result {
+enum stepper_event {
 	/** Steps set using move or set_target_position have been executed */
-	STEPPER_SIGNAL_STEPS_COMPLETED = 0,
-	STEPPER_SIGNAL_SENSORLESS_STALL_DETECTED = 1,
-	STEPPER_SIGNAL_LEFT_END_STOP_DETECTED = 2,
-	STEPPER_SIGNAL_RIGHT_END_STOP_DETECTED = 3,
+	STEPPER_STEPS_COMPLETED = 0,
+	/** Sensorless Stall detected */
+	STEPPER_SENSORLESS_STALL_DETECTED = 1,
```

Comment:
enums must match the namespace of the enum itself, since enums are "flat" in C, AKA bleed outside of the enum itself :)
```
STEPPER_SENSORLESS_STALL_DETECTED -> STEPPER_EVENT_SENSORLESS_STALL_DETECTED
```
also, does it matter that the stall detection is sensorless? this event could cover a bit more cases if just named:
```
STEPPER_EVENT_STALLED = 0
```

Reply:
When the end stop is hit, it's also a stall. Hence SensorLess Stall is required :-)

Reply:
The trinamic chips offer detection of end stops via 'event_stop_r', 'event_stop_l' when using reference switches. When its hits a hard-endstop, its a stall.

Reply:
Okay, i do get it, in application code a user would just want to detect stall, regardless of what kind of stall is i think. Hence it does make sense to drop the word "sensorless" @bjarki-andreasen and @dipakgmx. Thanks :-) it does make whole lot of sense what you guys are saying
 
So this is how it is:,
_There are several types of motor stall detection methods that are commonly used. Some of the main types include:

1. Current-based stall detection: This method relies on monitoring the motor's current to detect stall conditions. When a motor stalls, its current increases significantly. By measuring the current and comparing it to a predefined threshold, stall conditions can be detected. This method is often used in motor control systems.

2. Back-EMF-based stall detection: Back electromotive force (EMF) is generated in a motor when it rotates. In a stall condition, the back-EMF decreases or disappears. By monitoring the back-EMF, stall conditions can be detected. This method is commonly used in brushless DC motors.

3. Encoder-based stall detection: Encoders are devices that provide feedback about the position and speed of a motor. Stall detection can be achieved by monitoring the encoder signals. If the motor fails to rotate as expected, it indicates a stall condition. This method is frequently used in servo motors and high-precision applications.

4. Vibration-based stall detection: This method utilizes sensors to measure the vibrations produced by a motor. When a motor stalls, the vibration pattern changes significantly. By analyzing the vibration signals, stall conditions can be detected. This method is often used in industrial applications where precision is not critical.

It's important to note that the choice of motor stall detection method depends on the specific motor type, application requirements, and available sensor and control systems. Different methods may be used in combination to ensure accurate and reliable stall detection._